src/Winton.Extensions.Configuration.Consul/IConsulConfigurationSource.cs

```diff
@@ -23,6 +24,14 @@ public interface IConsulConfigurationSource : IConfigurationSource
         /// </summary>
         Action<ConsulClientConfiguration>? ConsulConfigurationOptions { get; set; }
 
+        /// <summary>
+        ///     Gets or sets a function taking a Consul <see cref="KVPair"/> to one or more
```

Comment:
I think it would be worth mentioning in this docstring that the output must be formatted in the same way as an `IConfiguration` object. E.g. the implementer is responsible for ensuring that the keys in the `KeyValuePair<string, string>` are properly formed with `:` separators etc. Maybe we should link to the MS docs on this.

Reply:
how does this sound:
```
    /// <summary>
    ///     Gets or sets a function taking a Consul <see cref="KVPair"/> to one or more key/value
    ///     pairs which are injected into the Microsoft <see cref="IConfiguration"/> system.
    /// </summary>
    /// <remarks>
    ///     The default ConvertConsulKVPairToConfig strategy is to remove the
    ///     <see cref="KeyToRemove"/> portion of the Consul Key and then apply the configured
    ///     <see cref="Parser"/> to parse the Consul value.  Note that if you customize this strategy
    ///     there are some requirements on the final format of
    ///     <see href="https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#configuration-keys-and-values">Configuration Keys and Values</see>
    /// </remarks>
```

Reply:
Yeah nice, I think there's just one slight amendment I'd make. I would change the first sentence to:

```
/// <summary>
    ///     Gets or sets a function which converts a Consul <see cref="KVPair"/> to one or more key/value
    ///     pairs which are injected into the Microsoft <see cref="IConfiguration"/> system.
/// </summary>
```

The rest of it sounds fine to me.

Reply:
Yeah nice, I think there's just one slight amendment I'd make. I would change the first sentence to:

```
/// <summary>
    ///     Gets or sets a function which converts a Consul <see cref="KVPair"/> to one or more key/value
    ///     pairs which are injected into the Microsoft <see cref="IConfiguration"/> system.
/// </summary>
```

The rest of it sounds fine to me.